src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThread.java

```diff
@@ -160,9 +195,13 @@ public StepExecution getStep() {
 
         try (Timeout timeout = Timeout.limit(5, TimeUnit.MINUTES)) {
             LOGGER.log(FINE, "runNextChunk on {0}", resumeValue);
-            Outcome o = resumeValue;
+            final Outcome o = resumeValue;
             resumeValue = null;
-            outcome = program.run0(o);
+            outcome = GroovyCategorySupport.use(Arrays.<Class>asList(CpsDefaultGroovyMethods.class, CpsDefaultGroovyMethodsExt.class, IteratorHack.class), new Closure<Outcome>(null) {
```

Comment:
Given the black magic here, it'd be nice to have some comments explaining what exactly is going on. =)